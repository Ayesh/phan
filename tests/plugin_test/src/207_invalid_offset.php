<?php

$arr = ['string' => 'z', '' => 'x', 0 => []];
function expect_string(string $x) { echo $x; }
expect_string($arr[true]);
expect_string($arr[3.5]);
expect_string($arr[STDIN]);
expect_string($arr[[]]);
expect_string($arr[null]);   // same as $arr['']
expect_string($arr[false]);  // same as $arr[0]
expect_string($arr[true]);
