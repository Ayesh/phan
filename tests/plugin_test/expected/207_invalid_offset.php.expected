src/207_invalid_offset.php:5 PhanTypeInvalidDimOffset Invalid offset true of $arr of array type array{string:'z',:'x',0:array{}}
src/207_invalid_offset.php:5 PhanTypeMismatchArgumentReal Argument 1 ($x) is $arr[true] of type null but \expect_string() takes string defined at src/207_invalid_offset.php:4
src/207_invalid_offset.php:6 PhanTypeInvalidDimOffset Invalid offset 3.5 of $arr of array type array{string:'z',:'x',0:array{}}
src/207_invalid_offset.php:6 PhanTypeMismatchArgumentReal Argument 1 ($x) is $arr[3.5] of type null but \expect_string() takes string defined at src/207_invalid_offset.php:4
src/207_invalid_offset.php:7 PhanPartialTypeMismatchArgument Argument 1 ($x) is $arr[STDIN] of type 'x'|'z'|array{} but \expect_string() takes string (array{} is incompatible) defined at src/207_invalid_offset.php:4
src/207_invalid_offset.php:7 PhanTypeMismatchDimFetch When fetching an array index from a value of type array{string:'z',:'x',0:array{}}, found an array index of type resource, but expected the index to be of type int|string
src/207_invalid_offset.php:8 PhanPartialTypeMismatchArgument Argument 1 ($x) is $arr[[]] of type 'x'|'z'|array{} but \expect_string() takes string (array{} is incompatible) defined at src/207_invalid_offset.php:4
src/207_invalid_offset.php:8 PhanTypeMismatchDimFetch When fetching an array index from a value of type array{string:'z',:'x',0:array{}}, found an array index of type array{}, but expected the index to be of type int|string
src/207_invalid_offset.php:10 PhanTypeMismatchArgumentReal Argument 1 ($x) is $arr[false] of type array{} but \expect_string() takes string defined at src/207_invalid_offset.php:4
src/207_invalid_offset.php:11 PhanTypeInvalidDimOffset Invalid offset true of $arr of array type array{string:'z',:'x',0:array{}}
src/207_invalid_offset.php:11 PhanTypeMismatchArgumentReal Argument 1 ($x) is $arr[true] of type null but \expect_string() takes string defined at src/207_invalid_offset.php:4
